instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/events/producer/deliver_messages.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module RubyKafka
+      module Events
+        module Producer
+          # The DeliverMessages module contains the instrumentation for producer deliver messages event
+          module DeliverMessages
+            extend self
+
+            EVENT_NAME = 'deliver_messages.producer.kafka'
+            SPAN_NAME = 'kafka.producer.deliver_messages'
+
+            def subscribe!
+              ::ActiveSupport::Notifications.subscribe('deliver_messages.producer.kafka') do |_name, start, finish, _id, payload|
+                attributes = { 'messaging.system' => 'kafka' }
```

Comment:
small nit but, as i think you're aware, there's a bunch of places in this instrumentation where we are frequently allocating strings that would be easier to reason about as constants.

Reply:
strings are frozen so we aren't allocating them each time.

IMHO constants makes things much harder to deal with for actual coding, you have to constantly refer to your constants file (pun intended)

Reply:
Just for further clarification you're referring to the attribute keys, correct?  For general developer ergonomics it's nice just be able to plunk in a string, but I guess "magic strings" can become a problem as the codebase grows.

FWIW, I'm fine leaving it as is or referencing a constants file if anyone has some really strong opinions on this.  

In lieu of a stronger opinion I will just leave it.

Reply:
There's some desire to add helpers for semantic conventions (see #109), but it isn't clear what the API for those helpers should look like. `OpenTelemetry::SDK::Resources::Constants` is one approach, and it is decidedly unpleasant to use. For now, I think the string literals are fine, but I really do hope that at some point someone will be inspired to define a reasonably pleasant set of semantic convention helpers and refactor all the auto instrumentation in terms of those helpers.

Reply:
+1 for using inline strings and not constants.